interface.js

```diff
@@ -0,0 +1,436 @@
+
+/* END Providers ------------------------------------------------------------ */
+
+/* ========================================================================== */
+/* Create Panes                                                               */
+/* ========================================================================== */
+
+// Maps a provider to each pane.
+// Future: will be configurable by the user
+let paneProviders = [Bard, OpenAi, Claude];
+
+// Create the panes based on the mapping
+paneProviders.forEach(provider => {
+
+	const providerPane = document.getElementById(`${provider.name.toLowerCase()}Pane`);
+	providerPane.classList.remove('hidden');
+
+});
+
+// add event listener for btn
+const promptEl = document.getElementById('prompt');
+
+// Submit prompt when the user presses Enter or Ctrl+Enter in the textarea input
+const SuperPromptEnterKey = store.get('SuperPromptEnterKey', false);
+
+// Submit the prompt by preccing Ctrl+Enter or Enter
+promptEl.addEventListener('keydown', function (event) {
+	const isCmdOrCtrl = event.metaKey || event.ctrlKey;
+	const isEnter = event.key === 'Enter';
+
+	if ((SuperPromptEnterKey && isEnter) || (isCmdOrCtrl && isEnter)) {
+		event.preventDefault();
+		document.getElementById('btn').click();
+	}
+});
+
+// Consolidated this code into the above event listener
+// Retaining it here in case there's a reason to revert
+//
+// promptEl.addEventListener('keydown', function (event) {
+// 	if (SuperPromptEnterKey) {
+// 		if (event.keyCode === 13) {
+// 			document.getElementById('btn').click();
+// 		} else {
+// 			if (event.keyCode === 13 && (event.metaKey || event.ctrlKey)) {
+// 				document.getElementById('btn').click();
+// 			}
+// 		}
+// 	}
+// });
+
+// promptEl.addEventListener('keydown', function (event) {
+// 	const isCmdOrCtrl = event.metaKey || event.ctrlKey;
+// 	if (isCmdOrCtrl && event.key === 'Enter') {
+// 		event.preventDefault();
+// 		form.dispatchEvent(new Event('submit'));
+// 	}
+// });
+
+// Sanitize input and send to all providers
+promptEl.addEventListener('input', function (event) {
+	const sanitizedInput = promptEl.value
+		.replace(/"/g, '\\"')
+		.replace(/\n/g, '\\n');
+
+	Bard.handleInput(sanitizedInput);
+	OpenAi.handleInput(sanitizedInput);
+	Claude.handleInput(Claude.webview, sanitizedInput);
+});
+
+const form = document.getElementById('form');
+
+// Submit prompt to all providers
+form.addEventListener('submit', function (event) {
+	const sanitizedInput = promptEl.value
+		.replace(/"/g, '\\"')
+		.replace(/\n/g, '\\n');
+	promptEl.value = '';
+	event.preventDefault();
+	Bard.handleSubmit(sanitizedInput);
+	OpenAi.handleSubmit(sanitizedInput);
+	Claude.handleSubmit(sanitizedInput);
+});
+
+// Adjust styling for enabled providers
+Bard.handleCss();
+OpenAi.handleCss();
+Claude.handleCss(Claude.webview);
+
+// fix double-pasting inside webviews
+OpenAi.setupCustomPasteBehavior();
+Bard.setupCustomPasteBehavior();
+Claude.setupCustomPasteBehavior();
```

Comment:
this feels like "setup code" that should belong higher up, because the other stuff directly above this is oninput and onsubmit code